frontend/ui/elements/screensaver_menu.lua

```diff
@@ -26,114 +26,144 @@ local function genMenuItem(text, setting, value, enabled_func, separator)
+                end,
                 callback = function()
-                    Screensaver:chooseFile(true)
+                    G_reader_settings:toggle("screensaver_show_message")
                 end,
+                separator = true,
             },
             {
-                text = _("Screensaver message"),
-                keep_menu_open = true,
+            text = _("Edit sleep screen message"),
+            enabled_func = function() 
+                if not G_reader_settings:isTrue("screensaver_show_message") then
+                    return false    
+                end
+            end,
+            keep_menu_open = true,
                 callback = function()
                     Screensaver:setMessage()
                 end,
             },
             {
-                text = _("Covers and images settings"),
+                text = _("Background fill"),
+                help_text = _([[This option will only become available, if you have selected 'Lock the screen in current state'
+                    as screensaver and have 'Sleep screen message' on.]]),
+                enabled_func = function() 
+                    if G_reader_settings:readSetting("screensaver_type") == "disable" and G_reader_settings:isTrue("screensaver_show_message") then
+                        return true    
+                    else
+                        return false
+                    end 
+                end,
                 sub_item_table = {
-                    genMenuItem(_("Black background"), "screensaver_img_background", "black"),
-                    genMenuItem(_("White background"), "screensaver_img_background", "white"),
-                    genMenuItem(_("Leave background as-is"), "screensaver_img_background", "none", nil, true),
-                    -- separator
-                    {
-                        text_func = function()
-                            local percentage = G_reader_settings:readSetting("screensaver_stretch_limit_percentage")
-                            if G_reader_settings:isTrue("screensaver_stretch_images") and percentage then
-                                return T(_("Stretch to fit screen (with limit: %1 %)"), percentage)
-                            end
-                            return _("Stretch to fit screen")
-                        end,
-                        checked_func = function()
-                            return G_reader_settings:isTrue("screensaver_stretch_images")
-                        end,
-                        callback = function(touchmenu_instance)
-                            Screensaver:setStretchLimit(touchmenu_instance)
-                        end,
-                    },
+                    genMenuItem(_("Black"), "screensaver_msg_background", "black"),
+                    genMenuItem(_("White"), "screensaver_msg_background", "white"),
+                    genMenuItem(_("Current state"), "screensaver_msg_background", "none", nil, true),
```

Comment:
Somehow, as we are in a submenu and don't see the context (that we read 1 second ago though :)), I find it fine as it was previously to remind of the context with "Black/White background" and "Leave background as-is".

Reply:
I will fight you on this one. Unless it's a goldfish using this, there is already plenty of context. Particularly with the greying out bit.

Reply:
On a slightly related note, `black/white background` is unproblematic for translation but `black/white` all by itself should be in [a context](http://koreader.rocks/doc/modules/gettext.html#pgettext).

Reply:
I am happy to use `black/white fill`. It is already inside a setting called `background fill`, there is plenty of context already, and to top it all up, it will not be available for most people anyway, until they activate two different options that is.